### PR TITLE
W-15188084: Encapsulate org.mule.runtime.core.privileged.util

### PR DESCRIPTION
**Pull request by elrodro83**

### CODE REVIEW

File: core-tests/src/test/java/org/mule/runtime/core/api/util/StringMessageUtilsTestCase.java [hunk 1]
```diff
@@ -93,33 +96,17 @@ public void toStringOnDoubleArrayShouldReturnStringRepresentation() {
...
 
     // the String will contain not more than exactly MAX_ARRAY_LENGTH elements
     String result = StringMessageUtils.toString(list);
-    assertTrue(result.endsWith("[..]]"));
+    assertThat(result, endsWith("[..]]"));
     assertEquals(StringMessageUtils.MAX_ELEMENTS - 1, countMatches(result, ","));
   }
 
-  @Test
-  public void testFormattedString() throws Exception {
```
Comment (eze210) [inline core-tests/src/test/java/org/mule/runtime/core/api/util/StringMessageUtilsTestCase.java:108]: Intentionally removed?

File: modules/spring-config/src/main/java/org/mule/runtime/config/internal/registry/CompositeOptionalObjectsController.java [hunk 2]
```diff
@@ -8,13 +8,13 @@
...
 import static org.mule.runtime.api.util.Preconditions.checkArgument;
 
-import org.mule.runtime.core.internal.util.ArrayUtils;
-
 import java.util.Collection;
 import java.util.List;
 
 import com.google.common.collect.ImmutableList;
 
+import org.apache.commons.lang3.ArrayUtils;
```
Comment (eze210) [inline modules/spring-config/src/main/java/org/mule/runtime/config/internal/registry/CompositeOptionalObjectsController.java:16]: Is this intentional?

Comment (elrodro83) [reply modules/spring-config/src/main/java/org/mule/runtime/config/internal/registry/CompositeOptionalObjectsController.java:16]: yes, this replaces `org.mule.runtime.core.internal.util.ArrayUtils` (removed up there, line 11)

File: core-tests/src/test/java/org/mule/runtime/core/api/util/StringMessageUtilsTestCase.java (hunk 1 above)
Comment (elrodro83) [reply core-tests/src/test/java/org/mule/runtime/core/api/util/StringMessageUtilsTestCase.java:108]: yes, method under test was removed: https://github.com/mulesoft/mule/pull/13291/files#diff-b52f0fa07360fbab33a6b8e6e2a110f159a49bd958e17f3c79d6af29a75de62eL42

**Review by eze210 — APPROVED** [review]

**Pull request merged**